roxie/udplib/udptrr.cpp

```diff
@@ -356,17 +363,21 @@ class CReceiveManager : implements IReceiveManager, public CInterface
 
         void okToSend(UdpSenderEntry *requester, unsigned slots)
         {
+            assert(slots > 0);
             switch (requester->state)
             {
             case flowType::request_to_send:
                 pendingRequests.remove(requester);
                 // Fall through
             case flowType::send_completed:
                 pendingPermits.append(requester);
+                requester->permitsIssued = allocatePermits(slots);
                 requester->state = flowType::ok_to_send;
                 break;
             case flowType::ok_to_send:
                 // Perhaps the sender never saw our permission? Already on queue...
+                returnPermits(requester->permitsIssued);
+                requester->permitsIssued = allocatePermits(slots);
```

Comment:
This means that the number of slots to allocated is decided without including the number that has just completed.  That could cause issues if large numbers have been allocated